plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/preferences/SSLCertificatesPreference.java

```diff
@@ -0,0 +1,175 @@
+	final static String SEPARATOR = ";";
+
+	static SSLCertificatesPreference instance = new SSLCertificatesPreference();
+
+	public static SSLCertificatesPreference getInstance() {
+		return instance;
+	}
+
+	private List<Item> savedItems = null;
+
+	private SSLCertificatesPreference() {}
+
+	protected IPreferenceStore getPreferenceStore() {
+		return OpenShiftUIActivator.getDefault().getCorePreferenceStore();
+	}
+
+	/**
+	 * Called by SSLCertificateCallback to retrieve the choice made in the dialog.
+	 * 
+	 * @param certificate
+	 * @return
+	 */
+	public Boolean getAllowedByCertificate(X509Certificate certificate) {
+		X509CertificateParser p = new X509CertificateParser(certificate);
+		Item i = new Item(false, p.getIssuer(), p.getValidity(), p.getFingerprint());
+		Item item = findItem(getSavedItems(), i);
+		return item != null ? item.checked : null;
+	}
+
+	/**
+	 * Called by SSLCertificateCallback to remember the choice made in the dialog.
+	 *  
+	 * @param certificate
+	 * @return
+	 */
+	public void setAllowedByCertificate(X509Certificate certificate, boolean result) {
+		X509CertificateParser p = new X509CertificateParser(certificate);
+		Item i = new Item(result, p.getIssuer(), p.getValidity(), p.getFingerprint());
+		List<Item> savedItems = getSavedItems();
+		Item item = findItem(savedItems, i);
+		if(item == null) {
+			savedItems.add(i);
+		} else {
+			item.checked = result;
+		}
+		saveItemsToPreference(savedItems);
+	}
+
+	/**
+	 * Called by preference page.
+	 * Each item is stored as  
+	 * 	checked SEPARATOR issuedBy SEPARATOR validity SEPARATOR fingerprint SEPARATOR
+	 * 
+	 * @return the single list instance, which is never replaced, only its content is changed
+	 */
+	List<Item> getSavedItems() {
+		if(savedItems == null) {
+			List<Item> savedItems = new ArrayList<>();
+			String value = getPreferenceStore().getString(ALLOWED_SERTIFICATES);
+			if(value != null && value.length() > 0) {
+				String[] tokens = value.split(SEPARATOR);
+				for (int i = 0; i + 4 <= tokens.length;) {
+					boolean checked = "true".equals(tokens[i++]);
+					String issuedBy = tokens[i++];
+					String validity = tokens[i++];
+					String fingerprint = tokens[i++];
+					savedItems.add(new Item(checked, issuedBy, validity, fingerprint));
+				}
+			}
+			this.savedItems = savedItems;
+		}
+		return savedItems;
+	}
+
+	private Item findItem(List<Item> items, Item item) {
+		return items.stream().filter(i -> i.equals(item)).findFirst().orElse(null);
+	}
+
+	/**
+	 * Called by preference page
+	 * @param items
+	 */
+	void saveWorkingCopy(List<Item> items) {
+		List<Item> savedItems = getSavedItems();
+		savedItems.clear();
+		savedItems.addAll(items);
+		saveItemsToPreference(savedItems);
+	}
+
+	private void saveItemsToPreference(List<Item> items) {
+		StringBuilder sb = new StringBuilder();
+		items.stream().forEach(i->sb.append(i.toPreferenceValue()));
+		getPreferenceStore().setValue(ALLOWED_SERTIFICATES, sb.toString());
```

Comment:
this doesn't seem to survive a workspace restart. I'm prompted to allow certificates after I restart eclipse.